web3_proxy/src/frontend/admin.rs

```diff
@@ -33,6 +39,117 @@ use std::sync::Arc;
 use time::{Duration, OffsetDateTime};
 use ulid::Ulid;
 
+/// `GET /admin/increase_balance` -- As an admin, modify a user's user-tier
+///
+/// - user_address that is to credited balance
+/// - user_role_tier that is supposed to be adapted
+#[debug_handler]
+pub async fn admin_increase_balance(
+    Extension(app): Extension<Arc<Web3ProxyApp>>,
+    TypedHeader(Authorization(bearer)): TypedHeader<Authorization<Bearer>>,
+    Query(params): Query<HashMap<String, String>>,
+) -> Web3ProxyResponse {
+    println!("Got here 1 {:?}", bearer);
+    let (caller, _) = app.bearer_is_authorized(bearer).await?;
+    let caller_id = caller.id;
+
+    println!("Got here 2, {:?}", caller_id);
+    // Establish connections
+    let db_conn = app
+        .db_conn()
+        .context("query_admin_modify_user needs a db")?;
+
+    // Check if the caller is an admin (if not, return early)
+    let admin_entry: admin::Model = admin::Entity::find()
+        .filter(admin::Column::UserId.eq(caller_id))
+        .one(&db_conn)
+        .await?
+        .ok_or(Web3ProxyError::AccessDenied)?;
+
+    // println!("Got here 2, {:?}", caller_id);
+    // Get the user from params
+    let user_address: Address = params
+        .get("user_address")
+        .ok_or_else(|| {
+            Web3ProxyError::BadRequest("Unable to find user_address key in request".to_string())
+        })?
+        .parse::<Address>()
+        .map_err(|_| {
+            Web3ProxyError::BadRequest("Unable to parse user_address as an Address".to_string())
+        })?;
+    let user_address_bytes: Vec<u8> = user_address.clone().to_fixed_bytes().into();
+    // Get the amount from params
+    let amount: Decimal = Decimal::from_str(params.get("amount").ok_or_else(|| {
```

Comment:
this works, but I usually keep it flatter by using `.map(Decimal::from_str)` instead.

Reply:
modified